skfuzzy/defuzzify/defuzz.py

```diff
@@ -86,10 +122,91 @@ def dcentroid(x, mfx, x0):
 
     """
     x = x - x0
-    u = (x * mfx).sum(axis=0).astype(float) / np.fmax(mfx.sum(axis=0),
-                                                      np.finfo(float).eps)
+    u = centroid(x,mfx)
     return x0 + u
 
+def bisector(x, mfx):
+    """
+    Defuzzification using bisector (`division of the area in two equally parts`) method.
+
+    Parameters
+    ----------
+    x : 1d array, length M
+        Independent variable
+    mfx : 1d array, length M
+        Fuzzy membership function
+
+    Returns
+    -------
+    u : 1d array, length M
+        Defuzzified result
+
+    See also
+    --------
+    skfuzzy.defuzzify.defuzz
+
+    """
+    '''
+    As we suppose linearity between each pair of points of x, we can calculate the exact area of the figure
+    (a triangle or a rectangle).
+    '''
+    sum_area = 0.0
+    accum_area = [0.0]*(len(x)-1)
+    #If the membership function is a singleton fuzzy set:
+    if len(x)==1:
+        return x[0]
+    #else return the sum of moment*area/sum of area
+    for i in range(1,len(x)):
```

Comment:
Another loop that begs for Cython. Don't feel any pressure to implement this, as it's a new dependency and outside the scope of this PR. But I think we'll want it moving forward.